src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -164,6 +179,23 @@ public void handleServiceChange(
                 (CacheConfiguration) wms.getRemoteResourcesCacheConfiguration().clone());
     }
 
+    private int getTimeoutMillis() {
+        String timeoutStr =
+                GeoServerExtensions.getProperty(GETMAPKVPREQUESTREADER_TIMEOUT_VARIABLE);
+        try {
+            if (timeoutStr != null) {
```

Comment:
Worth checking for emptiness like you do bellow?

Reply:
Replacing this with a common task method